e2e/bzlmod/BUILD.bazel

```diff
@@ -0,0 +1,123 @@
+load("@aspect_bazel_lib//lib:tar.bzl", "tar")
```

Comment:
Can we add the bzlmod example directly into `e2e/smoke` so that example is both BZLMOD and WORKSPACE?